packages/contracts/contracts/discovery/Batch.sol

```diff
@@ -4,32 +4,277 @@ pragma solidity =0.8.12;
 import {IBatch} from "./IBatch.sol";
 import {ICommon} from "./ICommon.sol";
 
+import {Math} from "../libraries/Math.sol";
+
+import "hardhat/console.sol";
+
 contract Batch is IBatch, ICommon {
-    uint256 batchId;
-    /// Ids of the projects
-    uint256[] projectIds;
-    /// number of approved projects in this batch
-    uint256 projectCount;
+    enum ProjectStatus {
+        InProgress,
+        Won,
+        Lost
+    }
+
+    struct Project {
+        address projectAddress;
+        ProjectStatus status;
+    }
+
+    struct Vote {
+        uint256 originalIndex;
+        uint32 count;
+    }
+
+    /// List of addresses for the project in the batch
+    address[] public projectAddresses;
+
+    /// Id of the batch
+    uint256 public batchId;
+
     /// number of available slots
-    uint256 slotCount;
-    Period votingPeriod;
+    uint256 public slotCount;
+
+    /// Period for which the batch is open for voting
+    Period public votingPeriod;
+
+    /// Address for the controller contract
+    address public controller;
+
+    /// Ordered list of slot winners
+    /// @dev only valid up to the last time there was a vote. Use getWinners() instead.
+    address[] winners;
+
+    /// Project address to extra information, such as status
+    /// @dev only valid up to the last time there was a vote. Use getProject(address) instead.
+    mapping(address => Project) projects;
```

Comment:
since we already have an address list, and other than that this struct only has the status, why not keep it as just a `mapping(address => ProjectStatus) statuses`?

Reply:
Not sure if we need anything else for a project. Didn't want to commit to only having status. Do you think we should?